python/ray/serve/multiplex.py

```diff
@@ -110,34 +116,43 @@ async def load_model(self, model_id: str) -> Any:
             raise ValueError("The model ID cannot be empty.")
 
         self.num_models.set(len(self.models))
-
         if model_id in self.models:
             # Move the model to the end of the OrderedDict to ensure LRU caching.
             model = self.models.pop(model_id)
             self.models[model_id] = model
+            return self.models[model_id]
         else:
-            # If the number of models per replica is specified, check if the number of
-            # models on the current replica has reached the limit.
-            if (
-                self.max_num_models_per_replica > 0
-                and len(self.models) >= self.max_num_models_per_replica
-            ):
-                # Unload the least recently used model.
-                self.models_unload_counter.inc()
-                unload_start_time = time.time()
-                await self.unload_model()
-                self.model_unload_latency_s.set(time.time() - unload_start_time)
-            # Load the model.
-            logger.info(f"Loading model '{model_id}'.")
-            self.models_load_counter.inc()
-            load_start_time = time.time()
-            if self.self_arg is None:
-                self.models[model_id] = await self._func(model_id)
-            else:
-                self.models[model_id] = await self._func(self.self_arg, model_id)
             self._push_multiplexed_replica_info = True
-            self.model_load_latency_s.set(time.time() - load_start_time)
-        return self.models[model_id]
+            self._model_load_tasks.add(model_id)
+            async with self._model_cache_lock:
+                # Check if the model has been loaded by another request.
+                if model_id in self.models:
+                    return self.models[model_id]
+
+                # If the number of models per replica is specified, check if
+                # the number of models on the current replica has reached the limit.
+                if (
+                    self.max_num_models_per_replica > 0
+                    and len(self.models) >= self.max_num_models_per_replica
```

Comment:
If the number of concurrent requests exceeds the max number of models, is it possible this could trigger the unload of a model that is actively in use?